### PR TITLE
Convert APR to appropriate rate for fine-grain compounding

### PR DESCRIPTION
**Pull request by Chris-Hibbert**
Had to rejigger a bunch of tests, using longer timeframes and larger
    borrowed amounts
Made manualTimer and scriptedPriceAuthority understand a timestep so
   we could do things by large fractions of a year

### CODE REVIEW

**Comment by Chris-Hibbert** [comment]
@dtribble, I think you asked for a guide to how to think about the numbers in the various tests, though I haven't been able to locate that conversation.

* `vault-contract-wrapper` needed to have larger time steps. This affects `test-vault` and`test-vault-interest`. I think I would get rid of both as the functionality they test is better covered elsewhere. They're a hassle to maintain.
* `test-vault-interest` needed loan and collateral multiplied by 100 or so or no interest would have been charged.
* in `test-stablecoin`, I changed the charging- and recording-periods for the tests that paid attention to interest to a week to make sense of the annual rate, and changed the manualTimer to move a day at a time. I'd have to boost the loan amounts for interest at 5%APR to be non-negligible when charged daily or weekly.  As it is, they're being charged single digit amounts of interest per day or week.
* In `test-interest`, daily interest on a loan of 100000n at 2.5% APR is 6n. These tests verify that the correct number of days of interest are charged as the calculate takes different parameters. There's at least one test case that makes the charging period be a month, and I verified that 203n, is the right charge for that.

File: packages/treasury/test/test-interest.js [hunk 1]
```diff
@@ -28,249 +33,283 @@ test('too soon', async t => {
 
 test('basic charge 1 period', async t => {
   const { brand } = makeIssuerKit('ducats');
+  const annualRate = makeRatio(250, brand, 10000n);
```
Comment (dtribble) [inline packages/treasury/test/test-interest.js]: use BASIS_POINTS constants

File: packages/treasury/test/test-interest.js [hunk 2]
```diff
@@ -28,249 +33,283 @@ test('too soon', async t => {
...
+    latestInterestUpdate: 0n,
+    interest: amountMath.make(0n, brand),
   };
-  t.deepEqual(calculator.calculate(debtStatus, 13n), {
-    latestInterestUpdate: 13n,
-    interest: amountMath.make(1000n, brand),
-    newDebt: amountMath.make(101000n, brand),
+  t.deepEqual(calculator.calculate(debtStatus, ONE_DAY), {
+    latestInterestUpdate: ONE_DAY,
+    interest: amountMath.make(6n, brand),
```
Comment (dtribble) [inline packages/treasury/test/test-interest.js:55]: Comment: 
6n is daily interest of 2.5% APR on 100k. The compounded interest is lost in the noise.

File: packages/treasury/test/test-interest.js [hunk 3]
```diff
@@ -28,249 +33,283 @@ test('too soon', async t => {
...
   const calculator = makeInterestCalculator(
     brand,
-    makeRatio((1n * SECONDS_PER_YEAR) / 3n, brand),
-    3n,
-    6n,
+    annualRate,
+    ONE_DAY,
+    ONE_MONTH,
   );
   const debtStatus = {
```
Comment (dtribble) [inline packages/treasury/test/test-interest.js:69]: comment: test interest between the 1st day and the 3rd day, which should be 6 (daily interest) * 2

File: packages/treasury/test/test-interest.js [hunk 4]
```diff
@@ -280,7 +319,7 @@ test('basic charge reasonable numbers', async t => {
...
     newDebt: amountMath.make(10000000n, brand),
-    interest: annualRate,
+    interest: amountMath.make(0n, brand),
```
Comment (dtribble) [inline packages/treasury/test/test-interest.js]: `amountMath.makeEmpy(brand)`?

File: packages/treasury/test/test-interest.js [hunk 5]
```diff
@@ -28,249 +33,283 @@ test('too soon', async t => {
...
   });
 });
 
 // 2.5 % APR charged weekly, charged daily
 test('basic charge reasonable numbers', async t => {
   const { brand } = makeIssuerKit('ducats');
   const annualRate = makeRatio(250, brand, 10000n);
-  const ONE_DAY = 60n * 60n * 24n;
-  const ONE_MONTH = ONE_DAY * 30n;
   const START_TIME = 1617734746n;
```
Comment (dtribble) [inline packages/treasury/test/test-interest.js:345]: Comment: This is a UNIX epoch time. 
Please say what the time is supposed to be.

Comment (dtribble) [reply packages/treasury/test/test-interest.js:345]: Tuesday April 6th 2021 at 11:45am PT

File: packages/treasury/test/test-interest.js [hunk 6]
```diff
@@ -295,31 +334,29 @@ test('basic charge reasonable numbers', async t => {
   });
   t.deepEqual(calculator.calculate(debtStatus, START_TIME + ONE_DAY), {
     latestInterestUpdate: START_TIME + ONE_DAY,
-    interest: amountMath.make(684n, brand),
-    newDebt: amountMath.make(10000684n, brand),
+    interest: amountMath.make(676n, brand),
```
Comment (dtribble) [inline packages/treasury/test/test-interest.js:371]: make this a magic constant?  676 is the daily * 100 with the compounding

File: packages/treasury/test/test-interest.js [hunk 7]
```diff
@@ -295,31 +334,29 @@ test('basic charge reasonable numbers', async t => {
...
     calculator.calculate(debtStatus, START_TIME + ONE_DAY + ONE_DAY),
     {
       latestInterestUpdate: START_TIME + ONE_DAY + ONE_DAY,
-      interest: amountMath.make(1368n, brand),
-      newDebt: amountMath.make(10001368n, brand),
+      interest: amountMath.make(1352n, brand),
+      newDebt: amountMath.make(10001352n, brand),
     },
   );
   // Notice that interest is compounding: 30 * 684 = 20520
```
Comment (dtribble) [inline packages/treasury/test/test-interest.js]: update comment for 676

File: packages/treasury/test/test-interest.js [hunk 8]
```diff
@@ -295,31 +334,29 @@ test('basic charge reasonable numbers', async t => {
...
   t.deepEqual(calculator.calculate(debtStatus, START_TIME + ONE_MONTH), {
     latestInterestUpdate: START_TIME + ONE_MONTH,
-    interest: amountMath.make(20555n, brand),
-    newDebt: amountMath.make(10020555n, brand),
+    interest: amountMath.make(20299n, brand),
+    newDebt: amountMath.make(10020299n, brand),
   });
 });
 
 // 2.5 % APR charged weekly, charged monthly
```
Comment (dtribble) [inline packages/treasury/test/test-interest.js]: fix comment

File: packages/treasury/test/test-interest.js [hunk 9]
```diff
@@ -329,7 +366,7 @@ test('basic charge reasonable numbers monthly', async t => {
   );
   const debtStatus = {
     newDebt: amountMath.make(10000000n, brand),
-    interest: annualRate,
+    interest: amountMath.make(0n, brand),
```
Comment (dtribble) [inline packages/treasury/test/test-interest.js]: `makeEmpty`

File: packages/treasury/test/test-stablecoin.js [hunk 10]
```diff
@@ -830,17 +830,15 @@ test('interest on multiple vaults', async t => {
...
   }
+  await waitForPromisesToSettle();
 
-  await manualTimer.tick();
-  await manualTimer.tick();
-  await manualTimer.tick();
   const aliceUpdate = await aliceNotifier.getUpdateSince();
   const bobUpdate = await bobNotifier.getUpdateSince();
-  const bobAddedDebt = 160n + 33n + 33n + 34n;
+  const bobAddedDebt = 160n + 3n;
```
Comment (dtribble) [inline packages/treasury/test/test-stablecoin.js:840]: comment: initial fee + 1/day (compounding is in the noise)

**Review by dtribble — CHANGES_REQUESTED** [review]
Mostly comment changes, but they are important for maintenance.

File: packages/treasury/test/test-interest.js [hunk 11]
```diff
@@ -10,6 +10,9 @@ import { makeInterestCalculator, SECONDS_PER_YEAR } from '../src/interest';
 const ONE_DAY = 60n * 60n * 24n;
 const ONE_MONTH = ONE_DAY * 30n;
 const ONE_YEAR = ONE_MONTH * 12n;
+const BASIS_POINTS = 10000n;
+const HUNDRED_THOUSAND = 100000n;
```
Comment (dtribble) [inline packages/treasury/test/test-interest.js:14]: BASIS_POINTS makes sense. The mere numeric ones are not clarifying as to their purpose. TEST_SCALE to make sure we have significant digits?

Comment (dtribble) [reply packages/treasury/test/test-interest.js:14]: Ah. I see below that TEN_MILLION is "LARGE_DEBT"? Perhaps HUNDRED_THOUSAND is MEDIUM_DEBT?

**Review by dtribble — APPROVED** [review]
I have 2 constant naming thoughts, but they aren't blocking

**Pull request merged**